heron/api/src/scala/com/twitter/heron/streamlet/scala/Streamlet.scala

```diff
@@ -74,7 +77,7 @@ trait Streamlet[R] {
     *
     * @param mapFn The Map Function that should be applied to each element
     */
-  def map[T](mapFn: R => _ <: T): Streamlet[T]
+  def map[T](mapFn: R => T): Streamlet[T]
```

Comment:
why was this change needed

Reply:
Scala Functions support `contravariant`(parameter type) and `covariance`(return type) type parameters as default as follows :

From `scala.Function1` source code:
```
trait Function1[-T1, +R] extends scala.AnyRef {
 this : scala.Function1[T1, R] =>
  def apply(v1 : T1) : R
  ...
}
```